### DISCUSSION TITLE
Help with setting up application access

### DISCUSSION THREAD

**Discussion by AritraWork97** [gravitational/teleport#7010] [Q&A]
Hi,

I have a jenkins server in a aws vpc with no public ip. I have a teleport proxy with a public ip and domain pointing to it. I have added the jenkins server as a teleport node with a tunnel and it works perfectly. When I try to add jenkins application to the mix, the application shows up on the application page on teleport but when I click on it the page times our. I would like some help so that I can access jenkins, which has a private ip only via teleport.

Any help will be appreciated.

**Comment by Valien** [comment]
Hey @AritraWork97 - look at the Teleport proxy logs and see what it is displaying (or what errors it is showing). Did you create an application token to add the Jenkins server to your proxy? It will need that and if you previously added a node token you'll need to create an app token.

Run: `tctl tokens ls` to show current tokens. Look for `App` in the `Type` column. If you do not have one that is active you can create one using the `tctl tokens add --type app` to create an `App` token.

**Reply by AritraWork97** [reply]
Hi,

I managed to solve it on my own. Before I was adding the jenkins app in the jenkins server itself rather than teleport proxy. After running the necessary commands it works perfectly.
